core/src/main/java/org/apache/calcite/sql/fun/SqlLibraryOperators.java

```diff
@@ -700,6 +727,14 @@ private SqlLibraryOperators() {
           ReturnTypes.TIMESTAMP_NULLABLE, OperandTypes.INTEGER,
           SqlFunctionCategory.TIMEDATE);
 
+  /** 2-argument form of the special minus-date operator
```

Comment:
I'm not sure how MINUS_DATE2 gets found by the validator. The syntax is 'special' so it isn't found by name. Users can't reference it in their queries. If the operator is only used in intermediate expressions move it to `class SqlInternalOperators`.

I also don't see why the return-type inference needs to be different.

Reply:
I parameterized the return-type inference because the standard MINUS_DATE operator has three arguments, and the return type is specified as ARG2. This is incompatible with BigQuery because there are only two arguments (the time/date/timestamp expression, and the interval being subtracted).